web/toolbar.js

```diff
@@ -85,6 +93,10 @@ class Toolbar {
       zoomOut: options.zoomOut,
     };
 
+    this.editorButtons = [
+      [AnnotationEditorType.FREE_TEXT, options.editorFreeTextButton],
+    ];
+
```

Comment:
Please follow the same pattern as in these methods here instead: https://github.com/mozilla/pdf.js/blob/19539676813b5df6ca4726a1bfa6765193b1067c/web/secondary_toolbar.js#L160-L162

Basically, you want to add a `#bindEditorModeListener`-method and please make sure that it implements the `toggled` *and* `aria-checked` attributes the same way as e.g. https://github.com/mozilla/pdf.js/blob/19539676813b5df6ca4726a1bfa6765193b1067c/web/secondary_toolbar.js#L219-L230